### ISSUE TITLE
Error in spirv-tile pass for linalg.generic with scf.if

### ISSUE THREAD

**Issue by stevesuzuki-arm** [iree-org/iree#11933]
### What happened?

Compilation error occurs in spirv-tile pass for linalg.generic with scf.if
```
error: expected to contain <= 1 scf.if ops
    %1 = linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel", "parallel"]} ins(%arg0 : tensor<128x384xi32>) outs(%0 : tensor<128x384xi32>) {
```

### Steps to reproduce your issue

```
#map = affine_map<(d0, d1) -> (d0, d1)>
module {
  func.func @linalg_generic_scf_if(%arg0: tensor<128x384xi32>) -> tensor<128x384xi32> {
    %c0_i32 = arith.constant 0 : i32
    %c2_i32 = arith.constant 2 : i32
    %c4096_i32 = arith.constant 4096 : i32
    %c1023_i32 = arith.constant 1023 : i32
    %0 = tensor.empty() : tensor<128x384xi32>
    %1 = linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel", "parallel"]} ins(%arg0 : tensor<128x384xi32>) outs(%0 : tensor<128x384xi32>) {
    ^bb0(%in: i32, %out: i32):
      %2 = arith.cmpi sle, %in, %c0_i32 : i32
      %3 = scf.if %2 -> (i32) {
        scf.yield %c0_i32 : i32
      } else {
        %4 = arith.cmpi sge, %in, %c4096_i32 : i32
        %5 = scf.if %4 -> (i32) {
          scf.yield %c1023_i32 : i32
        } else {
          %6 = arith.shrsi %in, %c2_i32 : i32
          scf.yield %6 : i32
        }
        scf.yield %5 : i32
      }
      linalg.yield %3 : i32
    } -> tensor<128x384xi32>
    return %1 : tensor<128x384xi32>
  }
}
```
Command:
`iree-compile --iree-hal-target-backends=vulkan-spirv`

### What component(s) does this issue relate to?

Compiler

### Version information

a76e13003

**Comment by stevesuzuki-arm** [comment]
Narrow-downed to `-iree-spirv-tile` pass

```
func.func @linalg_generic_scf_if_dispatch_0_generic_49152() {
  %c49152 = arith.constant 49152 : index
  %c0 = arith.constant 0 : index
  %c4096_i32 = arith.constant 4096 : i32
  %c1023_i32 = arith.constant 1023 : i32
  %c2_i32 = arith.constant 2 : i32
  %c0_i32 = arith.constant 0 : i32
  %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%c0) alignment(64) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<49152xi32>>
  %1 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) offset(%c0) alignment(64) : !flow.dispatch.tensor<readwrite:tensor<49152xi32>>
  %workgroup_id_x = hal.interface.workgroup.id[0] : index
  %workgroup_count_x = hal.interface.workgroup.count[0] : index
  %2 = affine.apply affine_map<()[s0] -> (s0 * 256)>()[%workgroup_id_x]
  %3 = affine.apply affine_map<()[s0] -> (s0 * 256)>()[%workgroup_count_x]
  scf.for %arg0 = %2 to %c49152 step %3 {
    %4 = flow.dispatch.tensor.load %0, offsets = [%arg0], sizes = [256], strides = [1] : !flow.dispatch.tensor<readonly:tensor<49152xi32>> -> tensor<256xi32>
    %5 = flow.dispatch.tensor.load %1, offsets = [%arg0], sizes = [256], strides = [1] : !flow.dispatch.tensor<readwrite:tensor<49152xi32>> -> tensor<256xi32>
    %6 = linalg.generic {indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>], iterator_types = ["parallel"]} ins(%4 : tensor<256xi32>) outs(%5 : tensor<256xi32>) attrs =  {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[256], [4]]>} {
    ^bb0(%in: i32, %out: i32):
      %7 = arith.cmpi sle, %in, %c0_i32 : i32
      %8 = scf.if %7 -> (i32) {
        scf.yield %c0_i32 : i32
      } else {
        %9 = arith.cmpi sge, %in, %c4096_i32 : i32
        %10 = scf.if %9 -> (i32) {
          scf.yield %c1023_i32 : i32
        } else {
          %11 = arith.shrsi %in, %c2_i32 : i32
          scf.yield %11 : i32
        }
        scf.yield %10 : i32
      }
      linalg.yield %8 : i32
    } -> tensor<256xi32>
    flow.dispatch.tensor.store %6, %1, offsets = [%arg0], sizes = [256], strides = [1] : tensor<256xi32> -> !flow.dispatch.tensor<readwrite:tensor<49152xi32>>
  }
  return
}
```
Command:

`iree-opt --iree-spirv-tile`

**Comment by stevesuzuki-arm** [comment]
Fixed by #11935
